Fix Railgun crash and minor projectile optimizations

reaperrr opened the pull request:
Fixes #16477 and does some minor perf/style/clean-up optimizations on several other projectiles.

reaperrr committed:
Minor projectile clean-ups and perf optimizations

reaperrr committed:
Fix DivideByZero crashes in Railgun projectile

pchote commented on OpenRA.Mods.Common/Projectiles/Railgun.cs:
```diff
@@ -153,14 +153,17 @@ void CalculateVectors()
 			// An easy vector to find which is perpendicular vector to forwardStep, with 0 Z component
 			LeftVector = new WVec(ForwardStep.Y, -ForwardStep.X, 0);
-			LeftVector = 1024 * LeftVector / LeftVector.Length;
+			if (LeftVector.Length != 0)
```
Using `LengthSquared` for these will save an unnecessary square root

reaperrr replied:
Done.

reaperrr commented:
From IRC:
> [01:10] <+pchote> reaperrr: can you put the sync attributes on their own line to reduce rebase conflicts with my stylecop pr?

reaperrr commented:
Updated.

> [01:10] <+pchote> reaperrr: can you put the sync attributes on their own line to reduce rebase conflicts with my stylecop pr?

To be honest, as far as I can tell in most places that would either have been counterproductive (in lines that auto-resolve should be able to handle), or wouldn't help (lines that would still conflict for different reasons, like inserted empty lines or removed [Sync] attribute etc.).

If you think there are more places that would really not conflict anymore if I updated them, please point them out.

pchote commented:
I was thinking more that I could then drop these files completely from #16532 and make that PR depend on this one.

reaperrr commented:
OK, updated.

reaperrr commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/de38ed83e427b6f1cb769fc15f09f6a48e20d846...473c64bf005f5192e4a3925578865c17d3aa5408
